setup.py

```diff
@@ -24,7 +24,7 @@
     "numpy<2.0",
     "scipy",
     "networkx",
-    "rustworkx",
+    "rustworkx>=0.14.0",
```

Comment:
Since the 0.15 was released just a month ago, we are trying to provide compatibility with a couple more versions. For instance, 0.14 was released on January 2024, so that at least a bit more flexibility to the users. @albi3ro